ecosystem/typescript/sdk_v2/src/crypto/ed25519.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import nacl from "tweetnacl";
+import { Deserializer } from "../bcs/deserializer";
+import { Serializer } from "../bcs/serializer";
+import { Hex } from "../core/hex";
+import { HexInput } from "../types";
+
+export class PublicKey {
```

Comment:
```suggestion
export class PublicKey extends Serializable {
```

I don't think it will work right now due to a conflict with the `toUint8Array` function (which I will change to match bcs.move and the rust/python sdk), but writing this here as a reminder that with these Serializable/Deserializable classes we should explicitly `extends Serializable` for the helper function that serializes and returns the bytes